src/item.cpp

```diff
@@ -1044,6 +1044,61 @@ static int get_base_env_resist( const item &it )
 
 }
 
+bool item::is_owned_by( const Character &c, bool available_to_take ) const
+{
+    // owner.is_null() implies faction_id( "no_faction" ) which shouldnt happen, or no owner at all.
+    // either way, certain situations this means the thing is available to take.
+    // in other scenarios we actaully really want to check for id == id, even for no_faction
+    if( owner.is_null() ) {
+        return available_to_take;
+    }
+    if( !c.get_faction() ) {
+        debugmsg( "item::is_owned_by() Character %s has no faction", c.disp_name() );
```

Comment:
`debugmsg` automatically adds the source file name and the line number, which is much more reliable than including the function name. Code used to get moved around / function are renamed, but the debug messages are rarely updated.

```suggestion
        debugmsg( "Character %s has no faction", c.disp_name() );
```